nixos/modules/services/web-apps/tachidesk-server.nix

```diff
@@ -0,0 +1,235 @@
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          A System Tray icon will be tried to be created.
+        '';
+      };
+
+      config = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          example = "127.0.0.1";
+          description = mdDoc ''
+            The ip that Tachidesk will bind to.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8080;
+          example = 4567;
+          description = mdDoc ''
+            The port that Tachidesk will listen to.
+          '';
+        };
+
+        basicAuth = {
+          enable = mkEnableOption (mdDoc ''
+            add a basic access authentication to Tachidesk-Server.
+            Enabling this option is useful when hosting on a public network/the Internet
+          '');
+
+          username = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = mdDoc ''
+              The username value that you have to provide when authenticating.
+            '';
+          };
+
+          passwordFile = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            example = "/var/secrets/tachidesk-server-password";
+            description = mdDoc ''
+              The password file containing the value that you have to provide when authenticating.
+            '';
+          };
+        };
+
+        downloadAsCbz = mkOption {
+          type = types.bool;
+          default = false;
+          description = mdDoc ''
+            Download chapters as .cbz files.
+          '';
+        };
+
+        localSourcePath = mkOption {
+          type = types.path;
+          default = cfg.dataDir;
+          defaultText = lib.literalExpression "tachidesk-server.dataDir";
+          example = "/var/data/local_mangas";
+          description = mdDoc ''
+            Path to the local source folder.
+          '';
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.submodule {
+          freeformType = with lib.types; attrsOf str;
+        };
+        default = { };
+        description = mdDoc ''
+          Extra configuration to write to {file}`server.conf`.
+          See <https://github.com/Suwayomi/Tachidesk-Server/wiki/Configuring-Tachidesk-Server> for more information.
+        '';
+        example = {
+          "server.socksProxyEnabled" = true;
+          "server.socksProxyHost" = "yourproxyhost.com";
+          "server.socksProxyPort" = "8080";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.config.port ];
+
+    users.groups = mkIf (cfg.group == "tachidesk") {
+      tachidesk = { };
+    };
+
+    users.users = mkIf (cfg.user == "tachidesk") {
+      tachidesk = {
+        group = cfg.group;
+        home = cfg.dataDir;
```

Comment:
I do not think a system user should have a home directory. As long as the datadir is readable and writable by the user, that's enough.

```suggestion
```

Reply:
Done